src/lib/storage/index/partial_hash/partial_hash_index_impl.cpp

```diff
@@ -134,14 +134,23 @@ size_t PartialHashIndexImpl<DataType>::estimate_memory_usage() const {
   bytes += sizeof(_positions);
   bytes += sizeof(_null_positions);
 
+  // Size of the keys.
   bytes += sizeof(std::pair<DataType, std::vector<RowID>>) * _positions.size();
 
   bytes += sizeof(std::vector<RowID>) * _positions.size();
-  bytes += sizeof(RowID) * std::distance(cbegin(), cend());
+
+  // Capacity for indexed RowIDs.
+  for (const auto& entry : _positions) {
```

Comment:
We can use structured bindings here: `[_, row_ids]` so that we can simply use `row_ids.capacity()` instead of `entry.second.capacity()`.

Reply:
I like!